packages/sdk-core/src/common/DelayHelper.ts

```diff
@@ -0,0 +1,22 @@
+export class Delay {
+    /**
+     * Promise set timeout wrapper.
+     * @param timeout timeout in ms
+     * @param signal abort signal
+     */
+    public static wait(timeout: number, signal?: AbortSignal) {
+        return new Promise<void>((resolve) => {
+            function abortCallback() {
+                clearTimeout(intervalId);
+                resolve();
```

Comment:
reject requires from you to capture exceptions from wait when we have a signal handler. In most cases we want to do it when we close the client. 

reject will add more headache that we don't want. For example cancelling all requests on application exit might throw an exception or will require from us to handle this exception somehow. 

I would prefer to stay with resolve. If the team is not ok with that, lets explore options

Reply:
So from what I see, you use this to retry, without the abort callback. If you would ever use abort there, this could mean that after cancelling the delay, sending would just continue without waiting. I'm not sure whether this is what you want.

Reply:
Let's simplify the logic for now. We don't need abort controller for now anywhere, so I will drop it from the class and we can bring it once again when we will need to start using it

Reply:
After thinking about it once again I'm changing it to reject instead